gems/pending/disk/modules/miq_disk_cache.rb

```diff
@@ -0,0 +1,113 @@
+require 'rufus/lru'
+require_relative "../MiqDisk"
+require 'ostruct'
+
+module MiqDiskCache
+  MIN_SECTORS_TO_CACHE = 32
+  DEF_CACHE_SIZE       = 100
+  DEBUG_CACHE_STATS    = false
+
+  attr_reader :d_size, :blockSize, :cache_size, :sectors_to_cache, :cache_hits, :cache_misses
+
+  def self.new(down_stream, d_info = nil)
```

Comment:
I actually originally implemented it that way then thought that perhaps using dInfo we leave room to expand the parameters we might want to tweak Cache Size (or LRU entries, as you note below) Sectors to Cache (or minimum sectors per cache entry) ...  could there be anything else? I originally thought of tweaking the block size but realized we  want to use the block size of the underlying disk module.